Add LAP SE 1.8 as a release build env

keadyk opened the pull request:
### Background

* Codifying changes made during draco-7_12 release process :)

### Purpose of Pull Request

* Add LAP SE1.8 environments as release builds for CTS/ATS-1
* (This seems useful given that LAP is now the default draco env on those machines.)
* [Fixes re-git issue #1363](https://re-git.lanl.gov/draco/draco/-/issues/1363)

### Description of changes

* Add functions to bash scripts to load LAP SE environments
* Small changes have already been made to corresponding draco module files to ensure the correct mpich is loaded.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash3/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

keadyk commented on tools/ats1-env.sh:
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
+      run "module load PrgEnv-intel"
+      run "module load lapse/1.8-intel"
+      # Be explicit about unloading haswell / loading the knl module!
+      # Something went awry when I tried to use the intel1904env-knl logic.
```
@KineticTheory Does this section seem okay? I originally followed your logic from the standard intel-19 module, but something went awry during my first attempt to release. The `module swap` didn't work properly, so the knl tag got appended to the build dir and the partition was set to knl... but the haswell module was still loaded.

This more explicit unload/load version worked, so I went with it. :)

KineticTheory replied:
See my comments above. I think the `unload` commands need to cover more possibilities.

KineticTheory commented on tools/ats1-env.sh:
```diff
@@ -24,7 +24,7 @@ else
 
 # The following toolchains will be used when releasing code
-environments="cce11env intel1904env intel1904env-knl"
+environments="cce11env intel1904env lapse18intelenv intel1904env-knl lapse18intelenv-knl"
```
I would remove the `intel1904env` entry here and the function definition.

keadyk replied:
Ok, sounds good. I wasn't super keen on the ballooning number of environments I was creating :)

KineticTheory commented on tools/ats1-env.sh:
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
+      sysname=$(/usr/projects/hpcsoft/utilities/bin/sys_name)
+      module use --append "/usr/projects/draco/Modules/${sysname}"
+
+      run "module unload draco"
```
`unload draco lapse`

KineticTheory commented on tools/ats1-env.sh:
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
+      module use --append "/usr/projects/draco/Modules/${sysname}"
+
+      run "module unload draco"
+      run "module unload PrgEnv-cray"
```
You can do something like this to be more generic:.  
`module unload PrgEnv-cray PrgEnv-intel PrgEnv-gnu`

keadyk replied:
Ok, thanks! :)

kendraplong committed:
Add LAP SE environment, dddress PR comments from KGT.

keadyk commented:
@KineticTheory I addressed your comments - thanks! I tested the env commands manually and everything is functioning. This is ready to merge when you are!
